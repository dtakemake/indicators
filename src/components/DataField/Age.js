import React from 'react'

const Age = React.forwardRef(({ ...props }, ref) => {

  return <span ref={ ref }>Возраст</span>
})

export default React.memo(Age)