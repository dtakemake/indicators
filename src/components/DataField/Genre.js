import React from 'react'

const Genre = React.forwardRef(({ ...props }, ref) => {

  return <span ref={ ref }>Пол</span>
})

export default React.memo(Genre)