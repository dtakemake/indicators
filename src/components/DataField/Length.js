import React from 'react'

const Length = React.forwardRef(({ length, setLength, unit }, ref) => {

  return <span ref={ ref }>Длина - { length } { unit }</span>
})

export default React.memo(Length)