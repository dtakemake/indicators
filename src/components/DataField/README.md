### папка с полями для ввода данных