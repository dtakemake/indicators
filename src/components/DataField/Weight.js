import React from 'react'

const Weight = React.forwardRef(({ weight, setWeight, unit }, ref) => {

  return <span ref={ ref }>Вес { weight } { unit }</span>
})

export default React.memo(Weight)