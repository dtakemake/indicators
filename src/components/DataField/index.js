import React from 'react'
import Age from './Age'
import Genre from './Genre'
import Length from './Length'
import Weight from './Weight'

const DataField = React.forwardRef(({ as: Component = 'div', ...props }, ref) => <Component { ...props } ref={ ref } /> )

DataField.Age = Age
DataField.Genre = Genre
DataField.Length = Length
DataField.Weight = Weight

export default DataField