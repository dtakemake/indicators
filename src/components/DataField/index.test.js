import React from 'react'
import ReactDOM from 'react-dom'
import TestRenderer from 'react-test-renderer'
import DataField from './'

describe('DataField', () => {
  
  it('renders without crashing', () => {
    let div = document.createElement('div')
    ReactDOM.render(<DataField className="DataFieldTest" />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders without crashing Age', () => {
    let div = document.createElement('div')
    ReactDOM.render(<DataField.Age className="DataFieldTestAge" />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders without crashing Genre', () => {
    let div = document.createElement('div')
    ReactDOM.render(<DataField.Genre className="DataFieldTestGenre" />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders without crashing Length', () => {
    let div = document.createElement('div')
    ReactDOM.render(<DataField.Length className="DataFieldTestLength" />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders without crashing Weight', () => {
    let div = document.createElement('div')
    ReactDOM.render(<DataField.Weight className="DataFieldTestWeight" />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('есть корректный снимок', () => {
    let component = TestRenderer.create(<DataField className="DataFieldTest" />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('есть корректный снимок', () => {
    let component = TestRenderer.create(<DataField.Age className="DataFieldTestAge" />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('есть корректный снимок', () => {
    let component = TestRenderer.create(<DataField.Genre className="DataFieldTestGenre" />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('есть корректный снимок', () => {
    let component = TestRenderer.create(<DataField.Length className="DataFieldTestLength" />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('есть корректный снимок', () => {
    let component = TestRenderer.create(<DataField.Weight className="DataFieldTestWeight" />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})