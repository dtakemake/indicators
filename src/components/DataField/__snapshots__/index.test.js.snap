// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataField есть корректный снимок 1`] = `
<div
  className="DataFieldTest"
/>
`;

exports[`DataField есть корректный снимок 2`] = `
<span>
  Возраст
</span>
`;

exports[`DataField есть корректный снимок 3`] = `
<span>
  Пол
</span>
`;

exports[`DataField есть корректный снимок 4`] = `
<span>
  Длина - 
   
</span>
`;

exports[`DataField есть корректный снимок 5`] = `
<span>
  Вес 
   
</span>
`;
